### PR TITLE
Add guide for new editor

### PR DESCRIPTION
**Pull request by argenos**
## Added

Guide for the GitHub.dev editor

## Checklist
- [x] before creating a new note, I searched the vault
- [x] new notes have the `.md` extension
- [x] (if applicable) attached images have descriptive file names
- [ ] (if applicable) for new notes in the folder "04 - Guides, Workflows, & Courses", I added a link to them in one of the "for {group X}" overviews: https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/%F0%9F%97%82%EF%B8%8F+04+-+Guides%2C+Workflows%2C+%26+Courses

### CODE REVIEW

**Comment by argenos** [comment]
@chrisgrieser @claremacrae @kometenstaub I didn't have much time to write this down and probably won't have much time to look at this till next month. Hope this is a good starting place, but feel free to edit and/or push to this branch if you have suggestions or improvements

**Comment by argenos** [comment]
From Discord, this line in CONTRIBUTING.md can be removed:
 
> ==Before submission, remember to revert the Community Vault to its basic settings, e.g. remove installed plugins, turn on spellchecking, and change any other unusual settings back to how they previously were.==

And we need to update CONTRIBUTING to point to this guide accordingly, and any other "for X" guides if they apply.

File: Editing notes using the github.dev editor.md
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:18]: Maybe worth linking to the Hub publish site and the Hub repo from these two bullet points... Many will not know what they mean, I feel.

File: Editing notes using the github.dev editor.md
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:17]: I'm guessing people need to have a GitHub account, and be logged in to GitHub in their browser, for this to work???

File: Editing notes using the github.dev editor.md
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:43]: The info in the pictures is great. I think we should review them after the page is published, for readability and accessibility.

And if we do re-do the images with smaller windows and large font sizes, maybe we could use files from the Hub, for context?

Here's an example of an existing page, on the Publish site, so show the type of problem:

image

**Review by claremacrae — COMMENTED** [review]
Many thanks for doing this!

I'll make a few suggested edits and push to the branch, each as separate commits...

I think it's worth merging this soon, but before too long, I think it will be worth redoing the images so that windows are smaller and the text in them is much larger - so that the text will be legible in the publish site. Especially when it's white text on a black background, which makes shrunk text even harder to read...

File: Editing notes using the github.dev editor.md
Comment (claremacrae) [reply Editing notes using the github.dev editor.md:18]: I've added those links.

File: Editing notes using the github.dev editor.md
Comment (claremacrae) [reply Editing notes using the github.dev editor.md:17]: I've added a note about that.

**Comment by claremacrae** [comment]
I've pushed a few suggested additions - and would be grateful for review.

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:15]: the guide "talks you through" sounds somewhat weird in my ears, but I am not sure how I'd rephrase it. Maybe sth like "will show you"?

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:31]: Like, I know that there is sometimes a reason for not adding everything to a commit when working with git in general. But for the occasional contributor not familiar with git, would there ever be a scenario where you do make changes that you do *not* end up adding to the commit?

I think it would be easier to simply phrase instruct people to simply add everything (or use such an add everything button, if there is one).

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:36]: for the less git experienced, I'd probably add/simplify this. Particularly for small contributions, the name the new branch gets is pretty much irrelevant, and the instruction "you can choose a new name" adds unnecessary complexity.

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:48]: aren't Pull requests strictly speaking git*hub* terminology? :P

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:61]: you might also add that the reviewer might request changes before approval

File: Editing notes using the github.dev editor.md
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:54]: for the less git experienced, it will be confusing why you have to describe your changes again, after you already described your changes above in the commit message 🤔

**Review by chrisgrieser — CHANGES_REQUESTED** [review]
I have added some comments, and while the guide in itself is very good, I am somewhat unsure about the purpose of recommending the github.dev editor to contributors.

Like this guide tells me *how* to use the github.dev editor, but it doesn't tell me *why* I should do it.

Like, using the normal GitHub Editor looks more intuitive to less tech-savvy users to me, and the other guide you wrote for that also contains far less references to git concepts (fork, branch, etc.). After reading your guide I am still asking myself, what the benefit of using the github.dev editor actually is. There are some things like having code editor commands when editing, but a less tech-savvy contributor might not be missing those anyway, but is confronted with a more complex editor with a lot more buttons and git lingo. 

And for the more tech-savvy contributors, using the githubdev editor means potentially learning a new tool when they are using a different code editor than VS Code (like me), as opposed to using the normal GitHub editor tech-savvy people are already familiar with. And having things like code editor commands isn't really a plus, since tech-savvy people that do more than mini-contributions would/could also simply use Obsidian (another tool they are already familiar with) and push the vault.

So I haven't followed the discussion you guys had a while back on the advantages of using the github.dev over normal github or over Obsidian, and I assume there are a bunch (otherwise you wouldn't recommend using it), but those benefits of using github.dev should be clearly emphasized at the beginning, so the reader knows why it's worth learning how to use github.dev.

**Comment by argenos** [comment]
I don’t think the text itself is necessary for all the images, it’s more about where the buttons are. I did zoom into the ones I thought mattered. 

I was not using the hub for most of the images because since I do have access to it, I won’t get all the prompts contributors get that do not have access to the hub do.

**Comment by argenos** [comment]
> I have added some comments, and while the guide in itself is very good, I am somewhat unsure about the purpose of recommending the github.dev editor to contributors.
> Like this guide tells me how to use the github.dev editor, but it doesn't tell me why I should do it.
> 
Off the top of my head:

Auto-closing of brackets and parenthesis. Editing multiple files, and easier upload of new ones. The file tree is more intuitive to add new files in the right folders, and makes it more noticeable when files don’t have an extension (e.g. the lack of .md). In the future, the ability to use markdownlint. Using preview in a second pane. 
> Like, using the normal GitHub Editor looks more intuitive to less tech-savvy users to me, and the other guide you wrote for that also contains far less references to git concepts (fork, branch, etc.).
> 
> After reading your guide I am still asking myself, what the benefit of using the github.dev editor actually is. There are some things like having code editor commands when editing, but a less tech-savvy contributor might not be missing those anyway, but is confronted with a more complex editor with a lot more buttons and git lingo.
> 

Those concepts are also in the previous guide. We can’t really escape git concepts when contributors are submitting new content through GitHub. It is also my intention to mention them explicit, because that’s needed vocabulary when we discuss changes, reviews or anything else also in Discord.

> And for the more tech-savvy contributors, using the githubdev editor means potentially learning a new tool when they are using a different code editor than VS Code (like me), as opposed to using the normal GitHub editor tech-savvy people are already familiar with. And having things like code editor commands isn't really a plus, since tech-savvy people that do more than mini-contributions would/could also simply use Obsidian (another tool they are already familiar with) and push the vault.
> 
Tech Savy contributors should just do a normal PR in this case. This guide isn’t really meant for them. If they still want to use the editor, I don’t see anything wrong with learning about vscode web.

**Comment by claremacrae** [comment]
> I don’t think the text itself is necessary for all the images, it’s more about where the buttons are. I did zoom into the ones I thought mattered. 

Understood. Personally, in a number of the whole-screen images I can't see the buttons or arrows pointing to them, when viewed at the sort of size they'll be published at... Others may have better eye site or lower res screens, of course.

> I was not using the hub for most of the images because since I do have access to it, I won’t get all the prompts contributors get that do not have access to the hub do.

Yes, I wondered about that. I was thinking of setting a second GitHub account so that I could make screenshots...

**Review by claremacrae — APPROVED** [review]
I'm approve this, because I feel that we should allow hub users to take advantage of the work that argenos has kindly done - and that the docs are definitely good enough to get people started with the new editing UI.

There will be time in future to make further additions and links to it, as we learn and get feedback from users.

And we can add more descriptive text about the pros and cons of the different editing approaches.

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [inline Editing notes using the github.dev editor.md:15]:
```suggestion
In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide shows you how to make the edit, using the "Edit in GitHub" button at the bottom of the note.
```

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:31]: In VSCode, there is a `+` button to add them all, I wouldn't go into too much detail which could be confusing.

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:36]: It says that also enter can be pressed, so that should be alright.

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:48]:
```suggestion
This is called creating a **Pull Request**.
```

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:54]: It already adds the title from the commit message. If a second or third line was used (I think not possible with this interface), then that would show up there.
And the contributor didn't describe what changes they made before, did they?

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [inline Editing notes using the github.dev editor.md:61]:
```suggestion
Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.
Reviewers may request changes before merging. 
```

**Review by kometenstaub — APPROVED** [review]
I didn't review the initial commits, I went through all files in their final version, so some comments I replied to might have been outdated.

File: Editing notes using the github.dev editor.md
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:51]: If need be, these images can also be opened at full size with right click, open in new window. As you said below, it is something we can improve later.

**Pull request merged**
